common/src/main/java/com/wynntils/screens/settings/widgets/FeatureSettingWidget.java

```diff
@@ -0,0 +1,189 @@
+import com.wynntils.screens.settings.WynntilsSettingsScreen;
+import com.wynntils.utils.MathUtils;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.TextComponent;
+
+public final class FeatureSettingWidget extends AbstractWidget {
+    private static final int MAX_RENDER_COUNT = 6;
+
+    private static final CustomColor BORDER_COLOR = new CustomColor(86, 75, 61, 255);
+    private static final CustomColor FOREGROUND_COLOR = new CustomColor(177, 152, 120, 255);
+    private static final CustomColor SCROLLBAR_COLOR = new CustomColor(137, 117, 92, 255);
+
+    private final WynntilsSettingsScreen settingsScreen;
+
+    private Feature cachedFeature = null;
+    private List<ConfigOptionElement> configWidgets = new ArrayList<>();
+    private int scrollIndexOffset = 0;
+
+    public FeatureSettingWidget(int x, int y, int width, int height, WynntilsSettingsScreen settingsScreen) {
+        super(x, y, width, height, new TextComponent("Feature Setting Widget"));
+
+        this.settingsScreen = settingsScreen;
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        recalculateConfigOptions();
+
+        poseStack.pushPose();
+        poseStack.translate(this.x, this.y, 0);
+
+        renderBackground(poseStack);
+
+        Feature selectedFeature = settingsScreen.getSelectedFeature();
+        if (selectedFeature == null) return;
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        selectedFeature.getTranslatedName(),
+                        0,
+                        this.width,
+                        8,
+                        this.width,
+                        CommonColors.WHITE,
+                        FontRenderer.TextAlignment.CENTER_ALIGNED,
+                        FontRenderer.TextShadow.OUTLINE);
+
+        renderConfigWidgets(poseStack, mouseX, mouseY, partialTick);
+
+        if (configWidgets.size() > MAX_RENDER_COUNT) {
+            renderScrollbar(poseStack);
+        }
+
+        poseStack.popPose();
+    }
+
+    private void renderConfigWidgets(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        final float padding = settingsScreen.width / 100f;
+
+        float renderY = settingsScreen.getBarHeight() + padding;
+
+        final float xOffset = this.width / 35f;
+
+        for (int i = scrollIndexOffset; i < configWidgets.size(); i += 2) {
+            ConfigOptionElement configWidgetLeft = configWidgets.get(i);
+            ConfigOptionElement configWidgetRight = configWidgets.size() > i + 1 ? configWidgets.get(i + 1) : null;
+            float renderHeight = this.height / 4f;
+
+            if (renderY + renderHeight > this.y + this.height) break;
```

Comment:
This can be simplified to ```java
            if (renderY > this.y + 3/4f * this.height) break;
```

Reply:
I am not sure this refactor is much simpler either.

Reply:
It gets rid of the `renderHeight` variable

Reply:
Well I need the renderHeight variable now anyways